tests/test_actions.py

```diff
@@ -0,0 +1,86 @@
+import os
+import pygame
+import pytest
+from unittest.mock import MagicMock
+from flatland.actions.actions import MovementMixin, SpeechMixin, LimbControlMixin
+from flatland.consts import Direction
+from typing import TYPE_CHECKING
+
+if TYPE_CHECKING:
+    from flatland.objects.base_objects import GameObject
+
+os.environ["SDL_VIDEODRIVER"] = "dummy"  # Use a headless display
+
+class Movement(MovementMixin):
```

Comment:
all the movement test below follow the same structure. hence, the code, as written, is redundant and shall be parametrized.

The best way to do it in pytest is like so:

```
@pytest.mark.parametrize(
    "initial_x, initial_y, direction, expected_x, expected_y",
    [
        (0, 0, Direction.UP,    0, -1),
        (0, 0, Direction.DOWN,  0, 1),
        (0, 0, Direction.LEFT, -1, 0),
        (0, 0, Direction.RIGHT, 1, 0),
    ]
)
def test_movement(initial_x, initial_y, direction, expected_x, expected_y):
    mover = Movement(initial_x, initial_y)
    mover.move(direction)
    assert mover.x == expected_x
    assert mover.y == expected_y
    assert mover.direction == direction
    
```